force-app/main/default/classes/AccountTriggerHandler.cls

```
@@ -0,0 +1,90 @@
+
+        //A method to change the account type to 'Prospect'(before insert)
+        //A method to copy the shipping address to the billing address.(before insert and update)
+        //A method to set the rating to 'Hot' if the Phone, Website, and Fax ALL have a value. (before insert and update)
+        //A method to create a contact related to the account with the following default values:(after insert)
+
+
+        // Validating CRUD
+
+
+        
+    Public static void beforeInsertSetType (List<Account> newAccounts) {
+        
+        //Going through new accounts without Type to set the Type field values
+        
+        for(Account acc: newAccounts) {
+
+            if (acc.Type == null) {
+                acc.Type = 'Prospect';
+            }
+
+        }
+
+    }
+
+
+    Public static void beforeInsertUpdateAddressCopy (List<Account> newAccounts) {
+        
+        //Going through new accounts without Type to set the Billing Address field values
+
+        for(Account acc : newAccounts) {
+
+            //if (acc.ShippingAddress != null) {
+                acc.BillingStreet = acc.ShippingStreet;
+                acc.BillingCity = acc.ShippingCity;
+                acc.BillingState = acc.ShippingState;
+                acc.BillingCountry = acc.ShippingCountry;
+                acc.BillingPostalCode = acc.ShippingPostalCode;
+            //}
+
+        }
+        
+    }
+
+
+    Public static void beforeInsertUpdateRating (List<Account> newAccounts) {
+        
+        //Going through new accounts without Type to set the Rating field values
+        for(Account acc: newAccounts) {
+
+            if (acc.Fax != null && acc.Phone != null && acc.Website != null) {
+                acc.Rating = 'Hot';
+
+            }
+
+        }
+
+    }
+
+
+    Public static void afterInsertContactCreate (List<Account> newAccounts) {
+
+        // Validate CRUD permissions for inserting Contact
+        if (!Schema.sObjectType.Contact.isCreateable()) {
```

Very nice! This is an example of what's called a "guard statement", whereby you "fail fast" and exit the method quickly if the operation can't be performed. Nice use of permission checking too.